packages/vite/src/node/plugins/importAnalysis.ts

```diff
@@ -848,6 +848,18 @@ export function createParseErrorInfo(
   }
 }
 
+const interopHelper = (module: any) => {
+  if (Array.isArray(module) || typeof module === 'string') {
+    return {
+      default: module,
+    }
+  }
+  if (module && module.__esModule) {
+    return module
+  }
+  return { ...module, default: module }
```

Comment:
Replying to your comment at https://github.com/vitejs/vite/issues/15542#issuecomment-1893421822, and regarding this, is there a reason you moved the array check above?

I was thinking doing something like this instead:

```js
   if (module && module.__esModule) {
     return module
   }
  if (typeof module === 'object' && !Array.isArray(module)) {
    return { ...module, default: module }
  }
  return { default: module }
```

So it also handles `module.export = true` and `module.export = 123` 🤔

Reply:
I have considered the boundary cases, and the `interopHelper` function should be able to handle basic data types properly.

The following examples all return `{ default: module }`
```js
interopHelper(true);
interopHelper(123);
interopHelper('123');
interopHelper([1, 2, 3]);
interopHelper(undefined);
interopHelper(null);
```
I think I can adopt the approach you provided, which seems more intuitive.